src/main/java/build/buf/protovalidate/internal/celext/CustomOverload.java

```diff
@@ -324,118 +359,62 @@ private static Overload isNan() {
+   *     otherwise.
    */
-  private static UnaryOp unaryOpForPrimitiveVal(Val val) {
-    switch (val.type().typeEnum()) {
+  private static Val uniqueList(Lister list) {
+    long size = list.size().intValue();
+    if (size == 0) {
+      return BoolT.True;
+    }
+    Set<Val> exist = new HashSet<>((int) size);
+    Val firstVal = list.get(IntT.intOf(0));
+    switch (firstVal.type().typeEnum()) {
       case Bool:
-        return uniqueMemberOverload(BoolT.BoolType, CustomOverload::uniqueScalar);
-      case Bytes:
-        return uniqueMemberOverload(BytesT.BytesType, CustomOverload::uniqueBytes);
-      case Double:
-        return uniqueMemberOverload(DoubleT.DoubleType, CustomOverload::uniqueScalar);
       case Int:
-        return uniqueMemberOverload(IntT.IntType, CustomOverload::uniqueScalar);
-      case String:
-        return uniqueMemberOverload(StringT.StringType, CustomOverload::uniqueScalar);
       case Uint:
-        return uniqueMemberOverload(UintT.UintType, CustomOverload::uniqueScalar);
+      case Double:
+      case String:
+      case Bytes:
+        break;
       default:
-        return Err::maybeNoSuchOverloadErr;
+        return Err.noSuchOverload(list, OVERLOAD_UNIQUE, null);
     }
-  }
-
-  /**
-   * Creates a custom unary operation overload for processing list values with a specific item type.
-   * The overload ensures that the list contains unique values of the specified item type.
-   *
-   * @param itemType The type of items expected in the list.
-   * @param overload The function to be invoked on the unique values.
-   * @return The {@link UnaryOp} instance for the unique member overload.
-   */
-  private static UnaryOp uniqueMemberOverload(
-      org.projectnessie.cel.common.types.ref.Type itemType, overloadFunc overload) {
-    return value -> {
-      Lister list = (Lister) value;
-      if (list == null || list.size().intValue() == 0L) {
-        return Err.noMoreElements();
-      }
-      Val firstValue = list.get(IntT.intOf(0));
-      if (firstValue.type() != itemType) {
-        return Err.newTypeConversionError(list.type(), itemType);
-      }
-      return overload.invoke(list);
-    };
-  }
-
-  @FunctionalInterface
-  private interface overloadFunc {
-    Val invoke(Lister list);
-  }
-
-  /**
-   * Determines if the input list of bytes contains unique elements. If the list contains duplicate
-   * byte arrays or strings, it returns false. If the list contains unique byte arrays or strings,
-   * it returns true.
-   *
-   * @param list The input list to check for uniqueness.
-   * @return {@link BoolT}.True if the list contains unique elements, {@link BoolT}.False otherwise.
-   */
-  private static Val uniqueBytes(Lister list) {
-    Set<Object> exist = new HashSet<>();
-    for (int i = 0; i < list.size().intValue(); i++) {
-      Object val = list.get(IntT.intOf(i)).value();
-      if (val instanceof byte[]) {
-        val = new String((byte[]) val, StandardCharsets.UTF_8);
```

Comment:
This was the source of a bug if the bytes contained invalid UTF-8. If so, the string would contain replacement characters for invalid bytes and two  different bytes values would return equal.